csds-client/client/client.go

```diff
@@ -0,0 +1,245 @@
+package client
+
+import (
+	csdspb "github.com/envoyproxy/go-control-plane/envoy/service/status/v2"
+	envoy_type_matcher "github.com/envoyproxy/go-control-plane/envoy/type/matcher"
+
+	"context"
+	"crypto/x509"
+	"errors"
+	"flag"
+	"fmt"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+	"google.golang.org/grpc/metadata"
+	"time"
+)
+
+type Flag struct {
+	uri         string
+	platform    string
+	authnMode   string
+	apiVersion  string
+	requestFile string
+	requestYaml string
+	jwt         string
+	configFile  string
+	monitorFreq time.Duration
+}
+
+type Client struct {
+	cc         *grpc.ClientConn
+	csdsClient csdspb.ClientStatusDiscoveryServiceClient
+
+	nm   []*envoy_type_matcher.NodeMatcher
+	md   metadata.MD
+	info Flag
+}
+
+// ParseFLags parses flags to info
+func ParseFlags() Flag {
+	uriPtr := flag.String("service_uri", "trafficdirector.googleapis.com:443", "the uri of the service to connect to")
+	platformPtr := flag.String("cloud_platform", "gcp", "the cloud platform (e.g. gcp, aws,  ...)")
+	authnModePtr := flag.String("authn_mode", "auto", "the method to use for authentication (e.g. auto, jwt, ...)")
+	apiVersionPtr := flag.String("api_version", "v2", "which xds api major version  to use (e.g. v2, v3 ...)")
+	requestFilePtr := flag.String("request_file", "", "yaml file that defines the csds request")
+	requestYamlPtr := flag.String("request_yaml", "", "yaml string that defines the csds request")
+	jwtPtr := flag.String("jwt_file", "", "path of the -jwt_file")
+	configFilePtr := flag.String("file_to_save_config", "", "the file name to save config")
+	monitorFreqPtr := flag.Duration("monitor_freq", 0, "the frequency of sending request in monitor mode (e.g. 500ms, 2s, 1m ...)")
+
+	flag.Parse()
+
+	f := Flag{
+		uri:         *uriPtr,
+		platform:    *platformPtr,
+		authnMode:   *authnModePtr,
+		apiVersion:  *apiVersionPtr,
+		requestFile: *requestFilePtr,
+		requestYaml: *requestYamlPtr,
+		jwt:         *jwtPtr,
+		configFile:  *configFilePtr,
+		monitorFreq: *monitorFreqPtr,
+	}
+
+	return f
+}
+
+// parseNodeMatcher parses the csds request yaml to nodematcher
```

Comment:
Maybe add a bit more mentioning that both file and command line yaml are parsed.

Reply:
Solved.